src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_gssapi.h"
+#include <assert.h>
+#include <string.h>
+
+static void NetSecurity_HandleError(uint32_t majorStatus, gss_buffer_t *outBufferHandle, uint32_t *outBufferLength)
+{
+    if (GSS_ERROR(majorStatus) || *outBufferHandle == nullptr || (*outBufferHandle)->length > UnsignedCast(std::numeric_limits<int>::max()))
+    {
+        uint32_t relBufferStatus;
+        gss_release_buffer(&relBufferStatus, *outBufferHandle);
+        *outBufferHandle = nullptr;
+        *outBufferLength = 0;
+    }
+    else
+    {
+        *outBufferLength = static_cast<uint32_t>((*outBufferHandle)->length);
```

Comment:
length is ```size_t```. When am using ```SignedCast```, I am getting the following error.

    pal_gssapi.cpp:47:28: error: implicit conversion loses integer precision:
      'typename std::make_signed<unsigned long>::type' (aka 'long') to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
        *outBufferLength = SignedCast((*outBufferHandle)->length);

That is why, I  am using static_cast. However, (line 41 above), am using static_cast only if the length is within the limits of uint32_t. The following will result in an error.

      (*outBufferHandle)->length > UnsignedCast(std::numeric_limits<int>::max()